src/AbstractPropelSqlExtractor.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace RebelCode\Migrations;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+use RebelCode\Migrations\PropelSqlExtractorInterface as I;
+use Traversable;
+
+/**
+ * Abstract functionality for a propel SQL migration extractor.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractPropelSqlExtractor
+{
+    /**
+     * Extracts the up and down SQL queries for a propel migration found at a given path.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $migration The path to the propel migration class file.
+     *
+     * @return array An array containing two keys mapping to sub-arrays of up and down SQL queries, keyed by schema ID.
+     *               The keys of the returned array are {@link K_UP_SQL} and {@link K_DOWN_SQL}.
+     */
+    protected function _extract($migration)
+    {
+        $code = $this->_getMigrationClassCode(get_class($migration));
```

Comment:
If `$migration` is truly a path, as its param docs say, in which case it should probably be called e.g. `$path`, then the `$code` is almost guaranteed to be `null`.

Reply:
The doc is mistaken then. The method accepts the migration object. I will amend.